aeron-driver/src/main/c/aeron_driver_conductor.c

```diff
@@ -3325,16 +3318,35 @@ aeron_rb_read_action_t aeron_driver_conductor_on_command(
 
         case AERON_COMMAND_REMOVE_PUBLICATION:
         {
-            aeron_remove_command_t *command = (aeron_remove_command_t *)message;
+            aeron_remove_publication_command_t *command = (aeron_remove_publication_command_t *)message;
 
-            if (length < sizeof(aeron_remove_command_t))
+            aeron_remove_publication_command_t complete_command;
+            aeron_remove_publication_command_t *complete_command_p;
+
+            if (length < sizeof(aeron_remove_publication_command_t))
             {
-                goto malformed_command;
+                if (length == 24) // old aeron_remove_command_t has no flags field
```

Comment:
This shouldn't be a magic number, consider using a constant or `offsetof(aeron_remove_publication_stct, flags)`.